File: AssemblyWithInterceptorAndFormattingWithThisInStaticMember/MethodTimeLogger.cs
using System;
using System.Collections.Generic;
using System.Reflection;

public static class MethodTimeLogger
{

    public static List<MethodBase> MethodBase = new List<MethodBase>();
    public static List<string> Messages = new List<string>();

    public static void Log(MethodBase methodBase, long milliseconds, string message)
    {
        Console.WriteLine(methodBase.Name + " " + milliseconds + ": " + message);

        MethodBase.Add(methodBase);

        if (message != null)
        {
            Messages.Add(message);
        }
    }

}

File: VBNetAssemblyWithInterceptor/MethodTimeLogger.vb
Imports System.Reflection

Public Module MethodTimeLogger
    Public MethodBase As List(Of MethodBase) = New List(Of MethodBase)()
    Public Messages As List(Of String) = New List(Of String)()

    Public Sub Log(method As MethodBase, milliseconds As Long, message As String)
        Console.WriteLine($"{method.Name} {milliseconds}")
        MethodBase.Add(method)

        If message IsNot Nothing Then
            Messages.Add(message)
        End If
    End Sub
End Module